Mixpanel/Mixpanel.m

```diff
@@ -254,9 +254,10 @@ + (void)assertPropertyTypesInArray:(NSArray *)arrayOfProperties depth:(NSUIntege
 
 - (NSString *)defaultDistinctId
 {    
-    NSString *distinctId = [self IFA];
+    NSString *distinctId;
 
 #if !defined(MIXPANEL_WATCHOS) && !defined(MIXPANEL_MACOS)
+    distinctId = [self IFA];
```

Comment:
Yes but right now the lines after `#endif` that check `if (!distinctId)` have different behavior because you no longer have it set to `[self IFA]` anymore if `MIXPANEL_WATCHOS` is defined. You could change it to initialize to `[self macOSIdentifier]` if it's macos and `[self IFA]` otherwise instead of inside the `!defined(MIXPANEL_WATCHOS)` block so that it will have the same behavior as before.